src/pages/config.tsx

```diff
@@ -84,10 +103,30 @@ function ConfigPage (): React.JSX.Element | null {
     window.parent?.postMessage({ source: 'helia-sw-config-iframe', target: 'PARENT', action: 'RELOAD_CONFIG', config }, {
       targetOrigin
     })
-    log.trace('config-page: RELOAD_CONFIG sent to parent window')
+    log.trace('RELOAD_CONFIG sent to parent window')
   }, [])
 
   useEffect(() => {
+    hasLocalGateway()
+      .then(async hasLocalGw => {
+        if (hasLocalGw) {
+          // check if local storage has it.
+          const unparsedGwConf = localStorage.getItem(LOCAL_STORAGE_KEYS.config.gateways)
+          const gwConf = unparsedGwConf != null ? JSON.parse(unparsedGwConf) as string[] : defaultGateways
+          if (!gwConf.includes(localGwUrl)) {
+            log(`Adding ${localGwUrl} to gateway list`)
+            gwConf.push(localGwUrl)
```

Comment:
nit: prepend to the list, so localhost is prioritized
```suggestion
            gwConf.unshift(localGwUrl)
```